tsinfer/formats.py

```diff
@@ -1778,17 +1820,23 @@ def add_site(
 
         if alleles is None:
             alleles = ["0", "1"]
+        if ancestral_state is None:
+            ancestral_allele_index = 0
+        elif ancestral_state == MISSING_DATA:
+            ancestral_allele_index = MISSING_DATA
+        else:
+            ancestral_allele_index = list(alleles).index(ancestral_state)
```

Comment:
What's the conversion to ``list`` for here? I think tuples also support the ``index`` operation.

Reply:
Happy to convert to tuple instead. Since `alleles` is passed in, it could be e.g. a numpy array, so I wanted to ensure art was something with an `index` operator.